proposals/function-pointers.md

```diff
@@ -164,27 +192,23 @@ unsafe class Util {
 ```
 
 In an unsafe context, a method `M` is compatible with a function pointer type `F` if all of the following are true:
-- `M` and `F` have the same number of parameters, and each parameter in `D` has the same `ref`, `out`, or `in` modifiers as the corresponding parameter in `F`.
+- `M` and `F` have the same number of parameters, and each parameter in `M` has the same `ref`, `out`, or `in` modifiers as the corresponding parameter in `F`.
 - For each value parameter (a parameter with no `ref`, `out`, or `in` modifier), an identity conversion, implicit reference conversion, or implicit pointer conversion exists from the parameter type in `M` to the corresponding parameter type in `F`.
 - For each `ref`, `out`, or `in` parameter, the parameter type in `M` is the same as the corresponding parameter type in `F`.
 - If the return type is by value (no `ref` or `ref readonly`), an identity, implicit reference, or implicit pointer conversion exists from the return type of `F` to the return type of `M`.
 - If the return type is by reference (`ref` or `ref readonly`), the return type and `ref` modifiers of `F` are the same as the return type and `ref` modifiers of `M`.
-- The calling convention of `M` is the same as the calling convention of `F`.
+- The calling convention of `M` is the same as the calling convention of `F`. This includes both the calling convention bit, as well as any calling convention flags specified in the unmanaged identifier.
```

Comment:
>This includes both the calling convention bit, as well as any calling convention flags specified in the unmanaged identifier. [](start = 78, length = 125)

I think we would want to expand on that. In particular, I think we should have a special handling of CallConvCDecl  modifier (and the like "special" legacy modifiers) in presence of a cdecl bit, as apposed to in presence of unmanaged bit (what ever the new bit we are going to use). One of the reasons, is that, I think, we would want to support overriding a method with a `cdecl with modifier` function pointer (C++ creates those) by a method with just `cdecl` function pointer in it. While looking for an override, we should make sure calling conventions match, once we find the matching override, the normal process will inherit the modifiers. Same goes for interface implementations.

Reply:
I see that you expanded on that in a section below. Looks reasonable.

---
In reply to: [448095465](https://github.com/dotnet/csharplang/pull/3441#discussion_r448095465) [](ancestors = 448095465)